queue.go

```diff
@@ -395,11 +395,19 @@ func (queue *redisQueue) consumerBatchConsume(batchSize int, timeout time.Durati
 		// debug(fmt.Sprintf("batch consume consume %d", len(batch))) // COMMENTOUT
 		consumer.Consume(batch)
 
-		// reset batch and timer
-		batch = batch[:0]
-		if !timer.Stop() {
-			<-timer.C
-		}
+		batch = batch[:0] // reset batch
+		stopTimer(timer)  // stop and drain the timer if it fired in between
+	}
+}
+
+func stopTimer(timer *time.Timer) {
```

Comment:
Well I'd hope there would be no such race condition. My assumption is that if you call `timer.Stop()` and then try to drain the channel after that, then there should be no way the timer can still fire after that. Otherwise I'd call that `Stop` broken.

There is a "race" that the timer still "fires" after `Stop` (when you observe it too late). But if you didn't observe it after stopping, then it should not happen after that.